From 7fa309db9dd5dac69079c1fb83b6f579555f7d53 Mon Sep 17 00:00:00 2001
From: Jeff Forcier <jeff@bitprophet.org>
Date: Fri, 3 Apr 2026 15:40:03 -0400
Subject: [PATCH] Update some classes to handle Invoke 3.x changes

Upstream: https://github.com/fabric/fabric/commit/7fa309db9dd5dac69079c1fb83b6f579555f7d53
Only the API compatibility changes are retained; unrelated restyling is omitted.

--- a/fabric/connection.py
+++ b/fabric/connection.py
@@ -217,6 +217,7 @@
         connect_timeout=None,
         connect_kwargs=None,
         inline_ssh_env=None,
+        remainder=None,
     ):
         """
         Set up a new object representing a server connection.
@@ -372,7 +373,7 @@
         # NOTE: parent __init__ sets self._config; for now we simply overwrite
         # that below. If it's somehow problematic we would want to break parent
         # __init__ up in a manner that is more cleanly overrideable.
-        super().__init__(config=config)
+        super().__init__(config=config, remainder=remainder)

         #: The .Config object referenced when handling default values (for e.g.
         #: user or port, when not explicitly given) or deciding how to behave.
--- a/fabric/tasks.py
+++ b/fabric/tasks.py
@@ -104,7 +104,5 @@
-    def make_context(self, config):
-        kwargs = self.init_kwargs
-        # TODO: what about corner case of a decorator giving config in a hosts
-        # kwarg member?! For now let's stomp on it, and then if somebody runs
-        # into it, we can identify the use case & decide how best to deal.
-        kwargs["config"] = config
+    def make_context(self, config, core_parse_result):
+        kwargs = dict(
+            self.init_kwargs, config=config, remainder=core_parse_result.remainder
+        )
         return Connection(**kwargs)
